### ISSUE TITLE
[FEAT] Nested filters in Hasura data provider

### ISSUE THREAD

**Issue by smparekh** [pankod/refine#2320]
**Is your feature request related to a problem? Please describe.**
Currently we can only filter on top level keys but we aren't taking advantage of the graphql query syntax by also allowing filtering via nested objects.

**Describe the solution you'd like**
We should allow querying for nested objects via the CrudFilter interface.

**Describe alternatives you've considered**
N/A

**Additional context**
A couple of implementation options.
1. New operator `nested`:
```tsx
{
  // empty field
  field: "",
  operator: "nested",
  value: {"category": {"type": {"_eq": "howto"}}},
},
{
  // or exclude field if operator nested
  operator: "nested"
  value: {"category": {"type": {"_eq": "howto"}}},
}
```

The goal being to construct a variables json object:
```json
{
  "where": {
    "category": {
      "type": {
        "_eq": "howto"
      }
    }
  }
}
```

**Comment by omeraplak** [comment]
Hey @smparekh ,
Thank you for contacting us! We'll discuss possible solutions with the core team and let you know 🚀

**Comment by omeraplak** [comment]
Hey @smparekh ,
We've released [`@pankod/refine-hasura@4.9.0`](https://github.com/pankod/refine/releases/tag/%40pankod%2Frefine-hasura%404.9.0). Could you approve this feature?
